### ISSUE TITLE
Show useful error messages in the cloud for too large files

### ISSUE THREAD

**Issue by jeroenpraat** [redmatrix/redmatrix#131]
_From @zzottel on August 6, 2014 8:58_

I just tried to upload a 24M video to my cloud using the web interface and ran into some problems that didn't show any useful error messages. If it's possible to throw good error messages at this point, it would be very nice if that was done.

1) If post_max_size in php.ini is set to a too small value (was 8M in this case), SabreDAV throws an XML error message that says it doesn't have a handler for this POST method.
2) If post_max_size is large enough (I set it to 100M), but upload_max_filesize is too small (default was 8M, too), Red just silently returns without doing anything or displaying an error message. I.e. the current cloud directory is shown, but with no new contents that should have been uploaded.

Additionally, setting the maximum image size on the admin page doesn't seem to have an effect on uploading a video. I'm not sure if this is intended, i.e. this size really only applies to images, so I just mention that. (Set it it to 10,000,000 and could upload the 24M video nevertheless.)

_Copied from original issue: friendica/red#552_

**Comment by jeroenpraat** [comment]
_From @ghost on August 8, 2014 14:39_

That particular error message is the failsafe "Something went wrong, God knows what" error from upstream.  It's what they spit out after they've run through a bunch of other stuff, and everything has failed.  Error reporting itself needs to be improved upstream before we can present something useful.  And, in fairness, it's quite tricky to find errors in foreign projects (note it's PHP/Apache, etc failing, not Sabre) where you don't now where the logs are - so this might be a wontfix.

On our end, we can do better, but not by much - we can put out an "Something went wrong" info box or something, but it's not immediately obvious how we'd manage to find an error that was actually useful without sending you off to your logs.

**Comment by jeroenpraat** [comment]
_From @zzottel on August 8, 2014 15:2_

Hm. I hoped there would be a method to find out how large the file to be uploaded is, and then check that size against those PHP variables before the actual upload begins, or have some callback that can cancel the upload in this case and display an error message or something like that.

**Comment by jeroenpraat** [comment]
_From @ghost on August 8, 2014 17:43_

Yeah, that was my initial thought too, but I'm working on the assumption that if something so obvious worked, they would have already done it upstream.  Maybe not, but it does sound like a chicken and egg problem.

**Comment by jeroenpraat** [comment]
_From @zzottel on August 8, 2014 18:23_

I _think_ Gallery3 has something like that. I'll try and see if I can find some code next week, but the project is a labyrinth.

**Comment by jeroenpraat** [comment]
_From @pafcu on August 10, 2014 8:0_

It seems like failure due to file size is a quite likely scenario (server defaults are often quite small), so why not suggest it as a possible source of failure, e.g. "Upload failed for an unknown reason. Maybe the maximum size of uploaded files is restricted on the server?". This would at least provide a hint where to look and how to fix it.

**Comment by jeroenpraat** [comment]
_From @hdi-kw on August 11, 2014 10:57_

Some weeks ago I made some tests with writing a SabreDAV plugin and own exceptions which hooks into the whole process quite early. It was to check against quotas and free disk limits. Got kicked off computer and internet since then and had no time to finish it. Haven't tried, but I think it should also be possible to check for these limits.

**Comment by jeroenpraat** [comment]
@zzottel This issue is closed because Redmatrix evolved into [Hubzilla](https://github.com/redmatrix/hubzilla). When this is still an issue in the most recent version of Hubzilla, please recreate this issue [over there](https://github.com/redmatrix/hubzilla/issues). We're sorry for any inconvenience this will maybe give you.
